### PR TITLE
Fully annotate `transform.py` and `transformations.py`

### PR DESCRIPTION
**Pull request by user27182**
### Overview

I ran `ruff check --select ANN` on both files and added all missing annotations.

### CODE REVIEW

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: pyvista/core/utilities/transform.py [hunk 1]
```diff
@@ -1454,13 +1462,13 @@ def inverse_matrix_list(self) -> list[NumpyArray[float]]:
         ]
 
     @property
-    def n_transformations(self) -> int:  # numpydoc ignore: RT01
+    def n_transformations(self: Transform) -> int:  # numpydoc ignore: RT01
         """Return the current number of concatenated transformations."""
         return self.GetNumberOfConcatenatedTransforms()
 
     @overload
     def apply(  # numpydoc ignore: GL08
```
Comment (user27182) [inline pyvista/core/utilities/transform.py:1470]:
```suggestion
    def apply(
```

File: pyvista/core/utilities/transform.py [hunk 2]
```diff
@@ -1938,5 +1952,5 @@ def check_finite(self) -> bool:  # numpydoc ignore: RT01
         return self._check_finite
 
     @check_finite.setter
-    def check_finite(self, value: bool) -> None:  # numpydoc ignore: GL08
+    def check_finite(self: Transform, value: bool) -> None:  # numpydoc ignore: GL08
```
Comment (user27182) [inline pyvista/core/utilities/transform.py:1955]:
```suggestion
    def check_finite(self: Transform, value: bool) -> None:
```

File: pyvista/core/utilities/transform.py [hunk 3]
```diff
@@ -1470,17 +1478,17 @@ def apply(  # numpydoc ignore: GL08
     ) -> NumpyArray[float]: ...
     @overload
     def apply(  # numpydoc ignore: GL08
```
Comment (user27182) [inline pyvista/core/utilities/transform.py:1480]:
```suggestion
    def apply(
```

File: pyvista/core/utilities/transform.py [hunk 4]
```diff
@@ -1470,17 +1478,17 @@ def apply(  # numpydoc ignore: GL08
...
+        obj: ConcreteDataSetType,
         /,
         *,
         inverse: bool = ...,
         copy: bool = ...,
         transform_all_input_vectors: bool = ...,
-    ) -> DataSet: ...
+    ) -> ConcreteDataSetType: ...
     @overload
     def apply(  # numpydoc ignore: GL08
```
Comment (user27182) [inline pyvista/core/utilities/transform.py:1490]:
```suggestion
    def apply(
```

File: pyvista/core/utilities/transform.py (hunk 1 above)
Comment (tkoyama010) [reply pyvista/core/utilities/transform.py:1470]: Good catch.

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**